src/lib/mongoBackend/MongoCommonRegister.cpp

```diff
@@ -269,6 +274,7 @@ static bool addTriggeredSubscriptions
       subs.insert(std::pair<string, TriggeredSubscription*>(subIdStr, trigs));
     }
   }
+  releaseMongoConnection(connection);
```

Comment:
At this point, cursor has been exahusted and I undersand that is safe to release connection. Not fully sure... maybe kill() should be called in the cursor as an extra security measure (asked for clarifications in the comments to the answer in http://stackoverflow.com/questions/33945987/thread-safeness-at-mongodb-c-driver-regarding-indirect-connection-usage-throug/33977828#33977828)